### ISSUE TITLE
[Bug][Types] 'lovefield.d.ts' is not a module

### ISSUE THREAD

**Issue by Luchillo** [google/lovefield#188]
This is what VSCode shows when importing lovefield, in my repo it shows `module '*'` if can't find, if i get into the dist folder then shows the error in the image.

![image](https://cloud.githubusercontent.com/assets/9698639/21508253/907ce0ce-cc4d-11e6-8128-f7e12f181eb6.png)

**Comment by Brooooooklyn** [comment]
you can try:
```bash
npm i @types/lovefield --save-dev
```

```ts
import * as lf from 'lovefield'

....
```

**Comment by Luchillo** [comment]
I did try the first one, when i was providing it as a global variable with Webpack's `ProvidePlugin` and it didn't work.

As for the second it worked very well, i realized later that if it was an UMD module i had to import like that, then the typings started working, however i still have the doubt, why does the one inside the lovefield package differs from the one in `DefinitelyTyped`? lack of update?

https://github.com/DefinitelyTyped/DefinitelyTyped/issues/13574

**Comment by andy-ms** [comment]
To bundle types, you should add [`"types": "./dist/lovefield.d.ts"`](http://www.typescriptlang.org/docs/handbook/declaration-files/publishing.html) to your `package.json`.
When types are bundled, we can remove `lovefield` from DefinitelyTyped and deprecate `@types/lovefield` in favor of just `lovefield`.

A few notes about improving the definition:

* Instead of using `/// <reference path="../es6-promise/es6-promise.d.ts"/>`, just require that people use [`--lib es6`](http://www.typescriptlang.org/docs/handbook/compiler-options.html).
* Instead of `declare module 'lf'`, add this at the top level to allow importing:
```ts
export = lf;
export as namespace lf;
```

**Comment by Luchillo** [comment]
@andy-ms What do you mean to use `--lib es6`?

**Comment by andy-ms** [comment]
The `--lib` compiler option controls what types are available by default. `--lib es6` includes (among others) types for `Promise`. (In `tsconfig.json` this would be `"lib": ["es6"]`.)

**Comment by Luchillo** [comment]
I meant to ask, is that an option in the library repo or is it in the client project that uses this lib?

For example i use webpack, and i don't recall this option being available.

**Comment by Luchillo** [comment]
Aside in my project i provide promises polyfill through `core-js` so i don't want to install `es6-promise` if not needed.

**Comment by andy-ms** [comment]
`--lib` is a TypeScript option, not a webpack option. Whatever build tool you use should give you the ability to specify TypeScript options.
When a library depends on ES6 things like `Promise`, generally it is the library user's responsibility to make a `Promise` shim (if necessary) and types available -- not the library's responsibility.

**Comment by Luchillo** [comment]
Yeah, i'm already making use of a polyfill that takes care of Promise's shimming, and most likely other users will.

**Comment by arthurhsu** [comment]
Closing this issue since there's no actionable item from Lovefield library.

**Comment by andy-ms** [comment]
Please re-open. The typings are currently unusable and they are easy to fix. Don't know why you think there is no actionable item since I mentioned a few. The most important one is to add [`"types": "./dist/lovefield.d.ts"`](http://www.typescriptlang.org/docs/handbook/declaration-files/publishing.html) to your `package.json`.

**Comment by arthurhsu** [comment]
Ok, I thought you were talking about adding to Luchillo's stuff (since the "your" looks like replying his post). TBH Lovefield team will not be able to test that npm package.json and we don't know if that would create side effects.
